BatteryConditionValidation.c

```diff
@@ -0,0 +1,71 @@
+/*****************************************************************************/
+/***    \file        BatteryConditionValidation.c
+***     \author      Abinesh Kanji Kovil Thatchinamoorthy (RBEI/ECK1) 
+***
+***     \brief       Source file for Battery Condition Monitoring
+/*****************************************************************************/
+
+/* Project Inlcudes */
+#include "BatteryConditionValidation.h"
+
+/* Funtion declarations */
+static int testBatteryCond_i(float temperature, float soc, float chargeRate);
+static int checkBatteryParam_i(float batteryParam_f ,float minRange_f, float maxRange_f,int batParamIndex_i);
+
+/*---------------------------------------------------------------------------*/
+/*     FUNCTION:    testBatteryCond_i
+ */
+/*!    \brief       check whether the Battery parameter is within the range
+ * 
+ *     \param       tempVal_f       [IN] - current Battery param value
+ *     \param       socVal_f        [IN] - acceptable minimum value
+ *     \param       chargeRateVal_f [IN] - acceptable maximum value
+ *     \returns     validity_i      [OUT]- Overall Battery Status
+ *
+*//*------------------------------------------------------------------------*/
+static int testBatteryCond_i(float tempVal_f, float socVal_f, float chargeRateVal_f) {
+  int validity_i = 0;
+  
+  validity_i |= (checkBatteryParam_i(tempVal_f      ,TEMP_VALID_MIN_VAL        ,TEMP_VALID_MAX_VAL       ,0));     /* Store the temperature status in last bit */
```

Comment:
Does checkBatteryParam_i return true when the battery requires attention? Or when it is ok? We need to see the implementation for knowing that. Is there any way you can make the function name clear?